includes/Core/Modules/Module.php

```diff
@@ -817,15 +816,42 @@ protected function exception_to_error( Exception $e, $datapoint ) {
 			if ( isset( $errors[0]['reason'] ) ) {
 				$reason = $errors[0]['reason'];
 			}
+		} elseif ( $e instanceof Google_Proxy_Code_Exception ) {
+			$access_code = $this->user_options->get( OAuth_Client::OPTION_PROXY_ACCESS_CODE );
+			$error_code  = $this->context->input()->filter( INPUT_GET, 'error', FILTER_SANITIZE_STRING );
+
+			if ( ! $error_code ) {
+				$error_code = $this->user_options->get( OAuth_Client::OPTION_ERROR_CODE );
+			}
+
+			if ( $error_code ) {
+				// Delete error code from database to prevent future notice.
+				$this->user_options->delete( OAuth_Client::OPTION_ERROR_CODE );
+			} elseif ( ! empty( $code ) ) {
+				$error_code = $code;
+			} else {
+				$error_code = $message;
+			}
+
+			$auth_client   = $this->authentication->get_oauth_client();
+			$message       = $auth_client->get_error_message( $error_code );
+			$status        = 401;
```

Comment:
Let's move the `$status` assignment closer to the top of the `catch` here since it is essentially constant for this case.

Reply:
Updated.